snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -109,6 +110,10 @@ public void initialize(String name, Map<String, String> properties) {
               + " JDBC driver to your jars/packages",
           cnfe);
     }
+
+    // Populate application identifier in jdbc client
+    properties.put("application", APP_IDENTIFIER);
```

Comment:
You might want to pull the `application` literal into a static final reference and add a link to the docs where it's used in snowflake since it's not a standard jdbc connector property.